tests/blackbox_test.go

```diff
@@ -184,28 +198,87 @@ func outer(t *testing.T, test types.Test, negativeTests bool) {
 		// Finish data setup
 		for _, part := range disk.Partitions {
 			if part.GUID == "" {
-				part.GUID = generateUUID(t, ctx)
+				part.GUID, err = generateUUID(t, ctx)
+				if err != nil {
+					return err
+				}
+			}
+			err := updateTypeGUID(t, part)
+			if err != nil {
+				return err
 			}
-			updateTypeGUID(t, part)
 		}
 
 		disk.SetOffsets()
 		for _, part := range test.Out[i].Partitions {
-			updateTypeGUID(t, part)
+			err := updateTypeGUID(t, part)
+			if err != nil {
+				return err
+			}
 		}
 		test.Out[i].SetOffsets()
 
 		// Creation
-		createVolume(t, ctx, i, disk.ImageFile, imageSize, 20, 16, 63, disk.Partitions)
-		disk.Device = setDevices(t, ctx, disk.ImageFile, disk.Partitions)
+		err = createVolume(t, ctx, i, disk.ImageFile, imageSize, 20, 16, 63, disk.Partitions)
+		if err != nil {
+			return err
+		}
+		// Move value into the local scope, because disk.ImageFile will change
+		// by the time this runs
+		imageFile := disk.ImageFile
+		defer func() {
+			err := os.Remove(imageFile)
+			if err != nil {
+				t.Errorf("couldn't remove %s: %v", imageFile, err)
+			}
+		}()
+
+		disk.Device, err = setDevices(t, ctx, disk.ImageFile, disk.Partitions)
+		if err != nil {
+			return err
+		}
+		device := disk.Device
+		defer func() {
+			err := destroyDevice(t, device)
+			if err != nil {
+				t.Errorf("couldn't destroy device: %v", err)
+			}
+		}()
 		test.Out[i].Device = disk.Device
-		rootMounted := mountRootPartition(t, ctx, disk.Partitions)
+
+		rootMounted, err := mountRootPartition(t, ctx, disk.Partitions)
+		if err != nil {
+			return err
+		}
+		partitions := disk.Partitions
+		defer func() {
```

Comment:
If `systemd.WaitOnDevice` fails in `mountRootPartition` then the mount point will be leaked